### PR TITLE
Optimize operator/active pattern checking in lookup items filtering

### PR DESCRIPTION
**Pull request by auduchinok**
For each group of same named items in completion lookup items filtering:

* don't needlessly map all items lists to get items to investigate
* use `vref.CoreDisplayName` for active pattern check instead of calculating compiled name

### CODE REVIEW

File: src/fsharp/service/ServiceDeclarationLists.fs [hunk 1]
```diff
@@ -615,21 +615,27 @@ type FSharpDeclarationListInfo(declarations: FSharpDeclarationListItem[], isForT
         // Filter out operators, active patterns (as values) and the empty list
         let items = 
             // Check whether this item looks like an operator.
-            let isOperatorItem(name, items: CompletionItem list) = 
-                match items |> List.map (fun x -> x.Item) with
-                | [Item.Value _ | Item.MethodGroup _ | Item.UnionCase _] -> IsOperatorName name
+            let isOperatorItem name (items: CompletionItem list) =
```
Comment (baronfel) [inline src/fsharp/service/ServiceDeclarationLists.fs]: the switch from a curried to a tupled argument representation means that a closure for `isOperatorItem name` will be created for each item in `items`, right?  Or am I thinking about this wrong?

Comment (auduchinok) [reply src/fsharp/service/ServiceDeclarationLists.fs]: It shouldn't, but I'll check.
It was a stylistic change.

Comment (baronfel) [reply src/fsharp/service/ServiceDeclarationLists.fs]: I extracted out this code to sharplab and it does seem like (at least in my extracted version) the helper functions are compiled as tuple-taking members, so this seems fine to me.

The sharplab sample can be seen [here](https://sharplab.io/#v2:DYLgZgzgNAJiDUAfA9gBwKYDsAEBlAnhAC7oC2AsAFBVH4bYBqATumNgLzYDe2AwsiwAiASwipgAQ3wA5CaXTYQ2Yk2GYA5tgC+NOgoCSJUh0YTgAVwXI2ANxZglzVtkTYAqpmHJMvCRAWuALLoRAAWyDAA4kzI5qgu2ABiyMi69Pyk4iFemIZkHFTc2HnGSiXaVFTAIcUQAPIYTBJEArLyyiZETJZVNaIN6E0tTOWYcgoAFMJGEEoZWUQ55cCiRACUBZTY29ikzQDGodjTZBDYAO7ToYU7rgDaJ6QAutgAtAB8Nzs7e0SHx0YAHTlS5hL7fBIlQEMMyWbAAfUhQOCYQi0Vi8URrihHhyvn8CLe71qAyGrXG2DG8nB31ciI+2DAZn84LpRMZzPQlUo1SItQAgvtFjZ0AAFZokJiYNoKM6cLo9Hl9CCC4ViiWDXJGbBTGZzZCZXlLbUrYgbdjg37/R5nUHXLa3bAPIwvD403YHI6PYHau3u7FAmEWBR2ZwM/QqoXCEXioiS6UU0NgQH8ISicRSGX+wkMpnAFkO7Zs3Oc7m87DoAAeciyANOJnBNuwrOJABlVoCwMJgJKdWBzDgJjB05IZOMoHXSBANgz3RD5ztMMg+VN6o1mgJysOxKOZZPp9gAGSH5uFhfn7ZLlfbjNj9qcABEdyeD42x+wAHoP9hTXyIOFzjOOJsD8EDsAAFT0EDMBgMDcW8fF0AnbxgHwZRzH+LBFloC4rjAiB8FIAAjZAVn2Slx2wIjzD5c4JEwPkWkZbte1iPlrGOBj0GAU0sALC8BMpZcdVEVVo3VONNXKG01jWIA==)

Comment (auduchinok) [reply src/fsharp/service/ServiceDeclarationLists.fs]: No, there's no additional allocation happens in Release here.

Before:
Screenshot 2020-03-12 at 17 58 50
After:
Screenshot 2020-03-12 at 17 56 24

**Pull request merged**